### DISCUSSION TITLE
Morgan Fingerprint differs between between 2020.09 and 2019.03

### DISCUSSION THREAD

**Discussion by JanoschMenke** [rdkit/rdkit#3614] [Q&A] [answered]
So I found that the my Morgan Fingerprints slightly changed after finally upgrading from 2019.03. I would assume that is due to the changes made in 2019.09. But I could not find the specific change that was made to the Morgan fingerprint.

>  The default values for some of the new fingerprint generators have been changed so that they more closely resemble the original fingerprinting code. In particular most fingerprinters no longer do count simulation by default and the RDKit fingerprint now sets two bits per feature by default.

My Question was something changed with regards to the morgan fingerprint, in that time and if so which specifc changes was made to the morgan fingerprinter. I could not find it.

**Comment by greglandrum** [comment, marked as answer]
To the best of my knowledge nothing changed in the morgan fingerprinting code itself. It's likely that what you're observing is caused by changes elsewhere in the code. If you share the structure of a molecule where you are seeing this happen, I can point out what changed.

**Reply by JanoschMenke** [reply]
Hey Greg, thanks for the quick reply. 

Two example molecules for which I obtained different fingerprints are:
`'C=CCOC(=O)c1c2c3ccc(SCC=C)c4cccc(c-2nc1=O)c43'`
`'c1ccc2c(c1)oc1cccc3cccc2c31'`

I process these with: 
`Chem.CanonSmiles(x)`
`PandasTools.AddMoleculeColumnToFrame(data, "smiles", "Molecule")`
`AllChem.GetMorganFingerprintAsBitVect(radius=2, bit_size=1024)`

It could be the change:

> Aromaticity perception of molecules like Cc1nnc2n1c1ccccc1n1c(C)nnc12 now correctly recognizes the full outer envelope, i.e. the bonds joining the rings are now also aromatic. In 2020.03 the three rings were connected by single bonds.

Which should result in a different encoding right?

**Reply by greglandrum** [reply]
It's probably not specifically that change.
You can see the difference here:
```
In [8]: print(rdkit.__version__)
2019.03.4

In [9]: Chem.CanonSmiles('c1ccc2c(c1)oc1cccc3cccc2c31')
Out[9]: 'c1ccc2c(c1)oc1cccc3cccc2c31'
```
```
In [14]: print(rdkit.__version__)
2020.09.1

In [15]: Chem.CanonSmiles('c1ccc2c(c1)oc1cccc3cccc2c31')
Out[15]: 'c1ccc2c(c1)Oc1cccc3cccc-2c13'
```
Notice that there's an extra explicit single bond in 2020.09.1. (it was there in 2020.03 as well).

**Reply by JanoschMenke** [reply]
Awesome thanks for the taking the time.
